utils/python/CIME/compare_test_results.py

```diff
@@ -0,0 +1,86 @@
+import CIME.compare_namelists, CIME.simple_compare
+from CIME.utils import expect
+from CIME.XML.machines import Machines
+from CIME.test_status import *
+from CIME.hist_utils import compare_baseline
+from CIME.case import Case
+
+import os, glob
+
+_MACHINE = Machines()
+
+###############################################################################
+def compare_history(testcase_dir_for_test, baseline_name):
+###############################################################################
+    with Case(testcase_dir_for_test) as case:
+        baseline_full_dir = os.path.join(case.get_value("BASELINE_ROOT"), case.get_value("COMPILER"), baseline_name, case.get_value("CASEBASEID"))
+        result, comments = compare_baseline(case, baseline_dir=baseline_full_dir)
+        if result:
+            return True, None
+        else:
+            logging.info(comments)
+            return False, "Diff'd"
+
+###############################################################################
+def compare_test_results(baseline_name, test_root, compiler, test_id=None, compare_tests=None):
+###############################################################################
+    test_id_glob = "*%s*%s*" % (compiler, baseline_name) if test_id is None else "*%s" % test_id
```

Comment:
I don't understand what you're doing here if test_id is None. In practice, though, it seems that test_id will always be provided; maybe just make this a required argument?